### PR TITLE
Updating artifact ProcessExecutions to add logic to persist audit rules

### PR DESCRIPTION
**Pull request by chajain**

### CODE REVIEW

File: artifacts/definitions/Linux/Events/ProcessExecutions.yaml [hunk 1]
```diff
@@ -17,20 +17,21 @@ required_permissions:
...
      // Install the auditd rule if possible.
-     LET _ <= SELECT * FROM execve(argv=[pathToAuditctl, "-a",
-          "exit,always", "-F", "arch=b64", "-S", "execve", "-k", "procmon"])
+     // Remove default filter if any
+     LET _ <= SELECT * FROM execve(argv=["/bin/bash", "-c", "if [ ! -f /etc/audit/rules.d/vrr_clear_default.rules ]; then echo '-d task,never -i' > /etc/audit/rules.d/vrr_clear_default.rules; chmod 0640 /etc/audit/rules.d/vrr_clear_default.rules; " + pathToAugenrules + " --load; fi"])
+
+     // add rule to collect new process 
+     LET _ <= SELECT * FROM execve(argv=["/bin/bash", "-c", "if [ ! -f /etc/audit/rules.d/vrr_procmon_b64.rules ]; then echo '-a always,exit -F arch=b64 -S execve -k vrr_procmon' > /etc/audit/rules.d/vrr_procmon_b64.rules; chmod 0640 /etc/audit/rules.d/vrr_procmon_b64.rules; " + pathToAugenrules + " --load; fi"])
+
+     LET _ <= SELECT * FROM execve(argv=["/bin/bash", "-c", "if [ ! -f /etc/audit/rules.d/vrr_procmon_b32.rules ]; then echo '-a always,exit -F arch=b32 -S execve -k vrr_procmon' > /etc/audit/rules.d/vrr_procmon_b32.rules; chmod 0640 /etc/audit/rules.d/vrr_procmon_b32.rules; " + pathToAugenrules + " --load; fi"])
```
Comment (guangyee) [inline artifacts/definitions/Linux/Events/ProcessExecutions.yaml]: maybe have both these rules in a single file, vrr_procmon.rules, as they are related?

File: artifacts/definitions/Linux/Events/ProcessExecutions.yaml [hunk 2]
```diff
@@ -17,20 +17,19 @@ required_permissions:
...
-     // Remove default filter if any
-     LET _ <= SELECT * FROM execve(argv=[pathToAuditctl, "-d", "task,never"])     
      // Install the auditd rule if possible.
-     LET _ <= SELECT * FROM execve(argv=[pathToAuditctl, "-a",
-          "exit,always", "-F", "arch=b64", "-S", "execve", "-k", "procmon"])
+     // Remove default filter if any
+     LET _ <= SELECT * FROM execve(argv=["/bin/bash", "-c", "if [ ! -f /etc/audit/rules.d/vrr_clear_default.rules ]; then echo '-d task,never -i' > /etc/audit/rules.d/vrr_clear_default.rules; chmod 0640 /etc/audit/rules.d/vrr_clear_default.rules; " + pathToAugenrules + " --load; fi"])
+
+     // add rule to collect new process 
+     LET _ <= SELECT * FROM execve(argv=["/bin/bash", "-c", "if [ ! -f /etc/audit/rules.d/vrr_procmon.rules ]; then echo -e '-a always,exit -F arch=b64 -S execve -k vrr_procmon\n-a always,exit -F arch=b32 -S execve -k vrr_procmon' >> /etc/audit/rules.d/vrr_procmon.rules; chmod 0640 /etc/audit/rules.d/vrr_procmon.rules; " + pathToAugenrules + " --load; fi"])
```
Comment (guangyee) [inline artifacts/definitions/Linux/Events/ProcessExecutions.yaml]: You don't need append `>>` one you are writing the file once. Just `>` should be fine.

Comment (jeffmahoney) [reply artifacts/definitions/Linux/Events/ProcessExecutions.yaml]: These rules assume that the rules will never change, especially if we place them all in one file. We really need to compare the contents (if any) of the existing file.

**Review by guangyee — APPROVED** [review]
Excellent! Thanks Charu!

**Comment by chajain** [comment]
Created PR https://github.com/SUSE/linux-security-sensor/pull/34

**Pull request closed without merging**